lib/daru/io/io.rb

```diff
@@ -202,23 +200,32 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
-        ::CSV
-          .read(path, 'rb',opts)
-          .tap { |c| yield c if block_given? }
-          .by_col.map { |col_name, values| [col_name, values] }.to_h
+        url = URI.parse(path)
+        if %w(http https).include?(url.scheme)
+          ::CSV
+            .parse(open(path), opts)
+            .by_col.map { |col_name, values| [col_name, values] }.to_h
+        else
+          ::CSV
+            .read(path, 'rb',opts)
+            .tap { |c| yield c if block_given? }
+            .by_col.map { |col_name, values| [col_name, values] }.to_h
+        end
```

Comment:
The whole idea of `open-uri` library is to uniformely open URIs and local paths. I believe, the code in lines 205-207 should be enough for both cases (local file and URI).